<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1366</width>
    <height>768</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true">/* Groupbox */

QGroupBox {
    border: 1px solid #DDD;
    border-radius: 9px;
    margin-top: 9px;
}

QGroupBox::title {
    subcontrol-origin: margin;
    left: 10px;
    padding: 0 3px 0 3px;
}</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>6</number>
    </property>
    <property name="topMargin">
     <number>6</number>
    </property>
    <property name="rightMargin">
     <number>6</number>
    </property>
    <property name="bottomMargin">
     <number>6</number>
    </property>
    <item>
     <widget class="QGroupBox" name="OptionsGB">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="minimumSize">
       <size>
        <width>220</width>
        <height>0</height>
       </size>
      </property>
      <property name="title">
       <string>Options</string>
      </property>
      <widget class="QPushButton" name="LoadOBJ">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>30</y>
         <width>181</width>
         <height>41</height>
        </rect>
       </property>
       <property name="text">
        <string>Load OBJ file</string>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
      </widget>
      <widget class="QGroupBox" name="MeshGroupBox">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>110</y>
         <width>201</width>
         <height>261</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QLabel" name="subdivTitleLabel">
          <property name="text">
           <string>Subdivision</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="stepsLayout">
          <item>
           <widget class="QLabel" name="SubdivLabel">
            <property name="text">
             <string>Steps:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSpinBox" name="SubdivSteps">
            <property name="enabled">
             <bool>true</bool>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QCheckBox" name="LimitPositionCheckBox">
          <property name="text">
           <string>Show Limit Positions</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="ShowSharpEdgesCheckBox">
          <property name="text">
           <string>Show Sharp Edges</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="ShowVerticesCheckBox">
          <property name="text">
           <string>Show Vertices</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="textBrowser">
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Sans Serif'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Select edges using right mouse button and vertices using ctrl and right mouse button&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="edgeInfoLayout">
          <item>
           <widget class="QLabel" name="EdgeInfoLabel">
            <property name="toolTip">
             <string>Right-click or Shift+Left-click on an edge to see its sharpness</string>
            </property>
            <property name="text">
             <string>Edge Sharpness:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QDoubleSpinBox" name="EdgeSharpness">
            <property name="decimals">
             <number>1</number>
            </property>
            <property name="minimum">
             <double>-1.000000000000000</double>
            </property>
            <property name="maximum">
             <double>10.000000000000000</double>
            </property>
            <property name="singleStep">
             <double>0.100000000000000</double>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="vertexInfoLayout">
          <item>
           <widget class="QLabel" name="VertexInfoLabel">
            <property name="toolTip">
             <string>Ctrl+Left-click on a vertex to see the number of sharp edges</string>
            </property>
            <property name="text">
             <string>Sharp Edges at Vertex:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="VertexSharpEdgeCountLabel">
            <property name="toolTip">
             <string>Ctrl+Left-click on a vertex to see the number of sharp edges</string>
            </property>
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QComboBox" name="MeshPresetComboBox">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>80</y>
         <width>181</width>
         <height>31</height>
        </rect>
       </property>
       <item>
        <property name="text">
         <string>-</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CreaseSquare</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CreaseCube</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>CreaseOctahedron</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Suzanne</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Spot</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>FancyCup</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>OpenCube</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>HollowTetra</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Brick</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>StackedCubesOpen</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>StackedCubesClosed</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>RockerArm</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>ItaPaperGeom</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Fertility</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>Fandisk</string>
        </property>
       </item>
      </widget>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2"/>
    </item>
    <item>
     <widget class="MainView" name="MainDisplay">
      <property name="mouseTracking">
       <bool>true</bool>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MainView</class>
   <extends>QOpenGLWidget</extends>
   <header>mainview.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
